server/src/main/java/org/apache/druid/metadata/SQLMetadataSegmentManager.java

```diff
@@ -744,6 +756,32 @@ public DataSegment map(int index, ResultSet r, StatementContext ctx) throws SQLE
 
     // Replace "dataSources" atomically.
     dataSources = newDataSources;
+    overshadowedSegments = ImmutableSet.copyOf(determineOvershadowedSegments(segments));
+  }
+
+  /**
+   * This method builds a timeline from given segments and finds the overshadowed segments
```

Comment:
nit: the comment refers to a single timeline but multiple timelines are being built